pkg/logs/parser/parser.go

```diff
@@ -21,3 +37,29 @@ type noopParser struct {
 func (p *noopParser) Parse(msg []byte) ([]byte, string, string, error) {
 	return msg, "", "", nil
 }
+
+// DecodingParser a generic decoding Parser
+type DecodingParser struct {
+	decoder *encoding.Decoder
+	Parser
+}
+
+// Parse does nothing for NoopParser
+func (p *DecodingParser) Parse(msg []byte) ([]byte, string, string, error) {
+	decoded, _, err := transform.Bytes(p.decoder, msg)
+	return decoded, "", "", err
+}
+
+// NewDecodingParser build a new DecodingParser
+func NewDecodingParser(e Encoding) *DecodingParser {
+	p := &DecodingParser{}
+	var enc encoding.Encoding
+	switch e {
+	case UTF16LE:
+		enc = unicode.UTF16(unicode.LittleEndian, unicode.UseBOM)
+	case UTF16BE:
+		enc = unicode.UTF16(unicode.BigEndian, unicode.UseBOM)
+	}
```

Comment:
The UTF-16 decoder from the `encoding/unicode` package, with the `unicode.UseBOM` does not systematically expect a BOM however it will override the chosen endianness if the input start with a BOM (And the BOM is stripped which is convenient).